keps/sig-node/127-user-namespaces/README.md

```diff
@@ -765,13 +838,34 @@ will rollout across nodes.
 
 ###### What specific metrics should inform a rollback?
 
+On Kubernetes side, the kubelet should start correctly.
+
+On the node runtime side, a pod created with pod.spec.hostUsers=false should be on RUNNING state if
+all node requirements are met. If the CRI runtime or the handler do not support the feature, the kubelet
+returns an error.
```

Comment:
This puts the pod into Failed state, right?

So the number of Failed pods is effectively what we can look at, right?

[Not a native metric, but something measurable]

Reply:
I've added the following to clarify. 

---

When a pod hits this error returned by the kubelet, the status in `kubectl` is shown as
`ContainerCreating` and the pod events shows:

```
  Warning  FailedCreatePodSandBox  12s (x23 over 5m6s)  kubelet            Failed to create pod sandbox: user namespaces is not supported by the runtime
```

The following kubelet metrics are useful to check:
 - `kubelet_running_pods`: Shows the actual number of pods running
 - `kubelet_desired_pods`: The number of pods the kubelet is _trying_ to run

If these metrics are very different, it means there are desired pods that can't be set to running.
If that is the case, checking the pod events to see if they are failing for user namespaces reasons
(like the errors shown in this KEP) is advised, in which case it is recommended to rollback or
disable the feature gate.